### DISCUSSION TITLE
Country Ban not work in d: redirect and Mute

### DISCUSSION THREAD

**Discussion by legendskyk** [inspircd/inspircd#2137] [Support]
### Description

We want to set up a country-based redirect on join to send users to a spam room and prevent abuse from malicious users. For example, using /mode #channelname +b d:#Spam:G:US worked in previous versions of InspIRCd (versions 2 and 3), but it seems to no longer work in version 4. Additionally, muting based on country, like /mode #channelname +m G:US, does not seem to work either.

### InspIRCd version

Inspircd 4.4.0

**Comment by sadiepowell** [comment]
>For example, using /mode #channelname +b d:#Spam:G:US worked in previous versions of InspIRCd (versions 2 and 3), but it seems to no longer work in version 4. 

That isn't a core feature. Its a third-party feature from inspircd-contrib.

>Additionally, muting based on country, like /mode #channelname +m G:US, does not seem to work either.

Thats because you're using it wrong. You want `+b m:G:US` or `+b mute:country:US` in the new style. This requires the muteban, geo_maxmind, and geoban modules to be loaded.

**Comment by legendskyk** [comment]
I did try /mode #channelname +b mute:country:US not worked still user doing chat even geomaxmind nd geoban both module are loaded.

**Reply by sadiepowell** [reply]
I can't replicate this on the InspIRCd Testnet.

```text
* Now talking on #testmute
* #testmute: mute:country:GB on Thu Dec 19 14:29:13 2024 by Sadie!sadie@inspircd.org
* #testmute :End of channel ban list.
<HexSadie> blah
* #testmute :You cannot send messages to this channel whilst a m: (mute) extban is set matching you.
```

Are you sure that the user you're trying to match:

1. Is not a server operator
2. Is not exempt because it has a channel prefix mode 
3. Is recognized as connecting from that country (this is shown in WHOIS). Geolocation isn't perfect and sometimes an IP isn't in the database.

Note that if you have `<muteban:notifyuser>` set to no/false the user will receive an echo as if their message was sent successfully. Additionally, clients that don't support echo message may show the message as if it was sent even though it was blocked. You should use another connection to validate whether the message actually went through.

**Reply by legendskyk** [reply]
sometime work sometime not work even when our operators put someone !m nick service make user mute like:

sets mode #channelname +b mute:*!*@ipaddress thats also not work user still do chat what bugs is that?

**Reply by sadiepowell** [reply]
It sounds like you don't have the right modules loaded. Can you pastebin the output of `/MODULES`?

**Reply by legendskyk** [reply]
there are all module installed here is pastebin see it:

https://pastebin.com/bVXMvyVA

your module seems have bug sometime its work sometime not work even our room operator mute abusers they still do msg even service put mute with right pattern.

**Reply by sadiepowell** [reply]
Do you have a server that I can test on? Because I've tried several things on the InspIRCd testnet and it seems to work for me.

**Reply by legendskyk** [reply]
Yes, we are running a web chat. Would you like to join our server?

**Reply by sadiepowell** [reply]
Yes.

**Reply by sadiepowell** [reply]
Actually, that won't be necessary as I think I can reproduce it now. I'll look into a fix.

**Reply by legendskyk** [reply]
so you found the bug?

**Reply by sadiepowell** [reply]
This should be fixed in insp4 git. Can you test?

**Reply by legendskyk** [reply]
you want me to reconfiguration module?

**Reply by sadiepowell** [reply]
You should upgrade to the latest insp4 git commit. This is an issue with acting extban matching rather than just the muteban module so you can't just update the module.

**Reply by legendskyk** [reply]
ok let me upgrade and let you know

**Reply by legendskyk** [reply]
I did upgrade still bugs on like this:

(28:11:00) * Zakoota sets mode: +b stripcolor:Bantwo!*@*

![image](https://github.com/user-attachments/assets/4261f48f-306d-4568-ae1b-5af24b4780c4)

 stripcolor ban not work
 
when we type !m then its get muted: but when we type !um mean unmute quiet commaand this its give error:

(28:11:03) -ChanServ- Unknown command unquiet #Mixchatroom Bantwo. "/msg ChanServ HELP" for help.

unquiet already installed in anope...

seems lots of bugs in this new version still on.

**Reply by sadiepowell** [reply]
>stripcolor ban not work

Can you confirm that you updated your entire server to the latest Git snapshot before testing? There have been some other fixes in Git since the last release relating to colour code stripping. Please re-pastebin the contents of /MODULES as a server operator.

> when we type !m then its get muted: but when we type !um mean unmute quiet commaand this its give error:

This is a response from services not from InspIRCd.

**Reply by legendskyk** [reply]
yes  6 hours ago i get git and upgrade it here is /module list:

https://pastebin.com/CsEaqmpp

stripcolors  not working

**Reply by sadiepowell** [reply]
Are you checking this from a different user? The user that sends a message that gets formatting codes stripped won't be able to see it. 

Also make sure they don't have a channel prefix rank and aren't an exempt operator.

**Reply by legendskyk** [reply]
yes strip colors user not have any rank just a register nick unregister nick colors being block but register user not

**Comment by sadiepowell** [comment]
This should be fixed in the latest release.
